src/Xamarin.Android.Build.Tasks/Tasks/CompileToDalvik.cs

```diff
@@ -106,26 +113,31 @@ protected override string GenerateCommandLineCommands ()
 
 			cmd.AppendSwitch (DxExtraArguments);
 
+			cmd.AppendSwitchIfNotNull ("--input-list=", inputListFile);
+
 			if (MultiDexEnabled) {
 				cmd.AppendSwitch ("--multi-dex");
 				cmd.AppendSwitchIfNotNull ("--main-dex-list=", MultiDexMainDexListFile);
 			}
 			cmd.AppendSwitchIfNotNull ("--output ", Path.GetDirectoryName (ClassesOutputDirectory));
 
-
-			// .jar files
-			if (AlternativeJarFiles != null && AlternativeJarFiles.Any ()) {
-				Log.LogDebugMessage ("  processing AlternativeJarFiles...");
-				cmd.AppendFileNamesIfNotNull (AlternativeJarFiles, " ");
-			} else {
-				Log.LogDebugMessage ("  processing ClassesOutputDirectory...");
-				var zip = Path.GetFullPath (Path.Combine (ClassesOutputDirectory, "..", "classes.zip"));
-				if (!File.Exists (zip)) {
-					throw new FileNotFoundException ($"'{zip}' does not exist. Please rebuild the project.");
+			using (var sw = new StreamWriter (path: inputListFile, append: false,
+					encoding: new UTF8Encoding (encoderShouldEmitUTF8Identifier: false))) {
+				// .jar files
+				if (AlternativeJarFiles != null && AlternativeJarFiles.Any ()) {
+					Log.LogDebugMessage ("  processing AlternativeJarFiles...");
+					foreach (var l in AlternativeJarFiles)
+						sw.WriteLine (l.ItemSpec);
+				} else {
+					Log.LogDebugMessage ("  processing ClassesOutputDirectory...");
+					var zip = Path.GetFullPath (Path.Combine (ClassesOutputDirectory, "..", "classes.zip"));
+					if (!File.Exists (zip)) {
+						throw new FileNotFoundException ($"'{zip}' does not exist. Please rebuild the project.");
+					}
+					sw.WriteLine (zip);
+					foreach (var jar in JavaLibrariesToCompile)
```

Comment:
Indentation/scope implies that you can't have both `JavaLibrariesToCompile` and `AlternativeJarFiles` at the same time.  Is this intentional?  Why?